portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -8215,31 +8216,83 @@ private Map<Locale, String> _getAlternateURLs(
 
 		for (Locale locale : availableLocales) {
 			String alternateURL = canonicalURL;
-			String alternateURLSuffix = canonicalURLSuffix;
+			String alternateURLSuffix = null;
 			String languageId = LocaleUtil.toLanguageId(locale);
 
 			if (replaceFriendlyURL) {
-				String friendlyURL = null;
+				String[] urlSeparators =
+					FriendlyURLResolverRegistryUtil.getURLSeparators();
 
-				for (LayoutFriendlyURL layoutFriendlyURL : layoutFriendlyURLs) {
-					if (!languageId.equals(layoutFriendlyURL.getLanguageId())) {
+				for (String urlSeparator : urlSeparators) {
+					if (!canonicalURLSuffix.startsWith(urlSeparator)) {
 						continue;
 					}
 
-					friendlyURL = layoutFriendlyURL.getFriendlyURL();
+					FriendlyURLResolver friendlyURLResolver =
+						FriendlyURLResolverRegistryUtil.getFriendlyURLResolver(
+							urlSeparator);
 
-					if (groupFriendlyURLPrefix != null) {
-						friendlyURL = groupFriendlyURLPrefix.concat(
-							friendlyURL);
-					}
+					HttpServletRequest httpServletRequest =
+						themeDisplay.getRequest();
 
-					break;
+					Map<String, Object> requestContext =
+						HashMapBuilder.<String, Object>put(
+							"request", httpServletRequest
+						).put(
+							WebKeys.LOCALE, locale
+						).build();
+
+					Map<String, String[]> params =
+						httpServletRequest.getParameterMap();
+
+					try {
+						LayoutFriendlyURLComposite layoutFriendlyURLComposite =
+							friendlyURLResolver.getLayoutFriendlyURLComposite(
+								themeDisplay.getCompanyId(),
+								themeDisplay.getScopeGroupId(), false,
+								canonicalURLSuffix, params, requestContext);
+
+						alternateURLSuffix =
+							layoutFriendlyURLComposite.getFriendlyURL();
+
+						break;
+					}
+					catch (PortalException portalException) {
+						if (_log.isDebugEnabled()) {
+							_log.debug(portalException);
+						}
+					}
```

Comment:
Is it safe to ignore errors here? Why not let the exception bubble up?

Reply:
@robertoDiaz what do you think about this suggestion?

Reply:
Ok, it makes sense. Can you forward this PR to @liferay-echo for a final review? This changes some of their components. Thanks!